controllers/notebook/notebook_controller.go

```diff
@@ -0,0 +1,506 @@
+			Name:          notebookv1alpha1.NotebookPortName,
+			ContainerPort: notebookv1alpha1.NotebookDefaultPort,
+		})
+	}
+}
+
+// set JUPYTER_ENABLE_LAB=yes in notebook container's env
+func setJuypterLabEnv(pod *corev1.Pod, notebook *notebookv1alpha1.Notebook) error {
+	if len(pod.Spec.Containers) == 0 {
+		err := fmt.Errorf("pod %s container list is zero", pod.Name)
+		return err
+	}
+	containers := pod.Spec.Containers
+	nbContainerExists := false
+	for i, container := range containers {
+		if container.Name == notebookv1alpha1.NotebookContainerName {
+			// add JUPYTER_ENABLE_LAB if not present
+			juypterLabEnvExists := false
+			for _, env := range container.Env {
+				if env.Name == "JUPYTER_ENABLE_LAB" {
+					juypterLabEnvExists = true
+					break
+				}
+			}
+			if !juypterLabEnvExists {
+				containers[i].Env = append(containers[i].Env, corev1.EnvVar{
+					Name:  "JUPYTER_ENABLE_LAB",
+					Value: "yes",
+				})
+			}
+
+			// add baseUrl and noToken in NOTEBOOK_ARGS
+			notebookArgsExists := false
+			for i, env := range container.Env {
+				if env.Name == "NOTEBOOK_ARGS" {
+					notebookArgsExists = true
+					nbEnv := generateNotebookArgsEnv(notebook)
+					container.Env[i].Value = nbEnv + " " + container.Env[i].Value
+					break
+				}
+			}
+
+			if !notebookArgsExists {
+				containers[i].Env = append(containers[i].Env, corev1.EnvVar{
+					Name:  "NOTEBOOK_ARGS",
+					Value: generateNotebookArgsEnv(notebook),
+				})
+			}
+
+			// add GRANT_SUDO
+			grantSudoExists := false
+			for _, env := range container.Env {
+				if env.Name == "GRANT_SUDO" {
+					grantSudoExists = true
+					break
+				}
+			}
+
+			if !grantSudoExists {
+				containers[i].Env = append(containers[i].Env, corev1.EnvVar{
+					Name:  "GRANT_SUDO",
+					Value: "yes",
+				})
+			}
+
+			// add JUPYTER_ALLOW_INSECURE_WRITES
+			JUPYTER_ALLOW_INSECURE_WRITES_EXISTS := false
+			for _, env := range container.Env {
+				if env.Name == "JUPYTER_ALLOW_INSECURE_WRITES" {
+					JUPYTER_ALLOW_INSECURE_WRITES_EXISTS = true
+					break
+				}
+			}
+
+			if !JUPYTER_ALLOW_INSECURE_WRITES_EXISTS {
+				containers[i].Env = append(containers[i].Env, corev1.EnvVar{
+					Name:  "JUPYTER_ALLOW_INSECURE_WRITES",
+					Value: "true",
+				})
+			}
+			nbContainerExists = true
+			break
+		}
+	}
+	if !nbContainerExists {
+		err := fmt.Errorf("pod %s doesn't have container named 'notebook', change the notebook container name to 'notebook'", pod.Name)
+		return err
+	}
+	return nil
+}
+
+// change the base url and pass in noToken
+func generateNotebookArgsEnv(notebook *notebookv1alpha1.Notebook) string {
+	nbServiceName := getNotebookServiceName(notebook)
+	ingressPath := path.Join("/", "notebooks", nbServiceName.Namespace, nbServiceName.Name)
+	baseUrl := fmt.Sprintf("--LabApp.base_url=%s --NotebookApp.base_url=%s --ServerApp.base_url=%s --ServerApp.allow_root=true", ingressPath, ingressPath, ingressPath)
+	noToken := "--LabApp.token=''"
```

Comment:
`noToken` indicates that notebook instance has no authentication?

Reply:
yeah, noToken for easier access